etcdmain/grpc_proxy.go

```diff
@@ -70,11 +73,28 @@ func newGRPCProxyStartCommand() *cobra.Command {
 	cmd.Flags().StringVar(&grpcProxyCert, "cert", "", "identify secure connections with etcd servers using this TLS certificate file")
 	cmd.Flags().StringVar(&grpcProxyKey, "key", "", "identify secure connections with etcd servers using this TLS key file")
 	cmd.Flags().StringVar(&grpcProxyCA, "cacert", "", "verify certificates of TLS-enabled secure etcd servers using this CA bundle")
+	cmd.Flags().BoolVar(&grpcProxySelfAddr, "self-addr", false, "true to replace members in member list response")
```

Comment:
I meant, after removing `self-addr` flag, it will default to returning either listen client url for proxy or other registered proxy addressed, from member list API.